stdlib/src/blackbox_fallbacks/utils.rs

```diff
@@ -0,0 +1,72 @@
+use acir::{
+    circuit::{
+        directives::Directive,
+        opcodes::{BlackBoxFuncCall, FunctionInput},
+        Opcode,
+    },
+    native_types::{Expression, Witness},
+    FieldElement,
+};
+
+use crate::helpers::VariableStore;
+
+fn round_to_nearest_mul_8(num_bits: u32) -> u32 {
+    let remainder = num_bits % 8;
+
+    if remainder == 0 {
+        return num_bits;
+    }
+
+    num_bits + 8 - remainder
+}
+
+pub(crate) fn round_to_nearest_byte(num_bits: u32) -> u32 {
+    round_to_nearest_mul_8(num_bits) / 8
+}
+
+// TODO: Maybe this can be merged with `bit_decomposition`
+pub(crate) fn byte_decomposition(
+    gate: Expression,
+    num_bytes: u32,
+    mut num_witness: u32,
```

Comment:
```suggestion
    num_witness: mut& u32,
```

Thoughts on passing a mutable reference here? We've got a _lot_ of extra return values everywhere due to needing to update `num_witness`.

Reply:
I think we can instead just pass the VariableStore as a mutable reference. See comment ontop of bit_decomposition as to why we did not do it initially.